geoapps/utils/plotting.py

```diff
@@ -541,38 +542,73 @@ def check_data_type(data):
     return data
 
 
-def plot_convergence_curve(h5file):
-    """"""
-    workspace = Workspace(h5file)
-    names = [group.name for group in workspace.groups if isinstance(group, SimPEGGroup)]
-    objects = widgets.Dropdown(
+def plot_convergence_curve(outfile: str) -> widgets.interactive | None:
+    """
+    Plot the convergence curve from a SimPEG *.out file.
+
+    :param outfile: Full path to the SimPEG *.out file
+
+    :return fig: widgets.interactive figure.
+    """
+
+    if not path.exists(outfile):
+        warnings.warn(f"File {outfile} does not exist.")
+        return
+
+    with open(outfile, encoding="utf-8") as f:
+        lines = f.readlines()
+
+    names = lines[0].split(" ")[:-1]
+    data = []
+    for line in lines[1:]:
+        data += [[string_to_numeric(k) for k in line.split(" ")[:-1]]]
+
+    data = np.vstack(data)
+    result = dict(zip(names, data.T))
+    curve_a = widgets.Dropdown(
+        options=names,
+        value=names[2],
+        description="inversion Group:",
+        style={"description_width": "initial"},
+    )
+    curve_b = widgets.Dropdown(
+        options=names,
+        value=names[3],
+        description="inversion Group:",
+        style={"description_width": "initial"},
+    )
+    x_axis = widgets.Dropdown(
         options=names,
         value=names[0],
         description="inversion Group:",
         style={"description_width": "initial"},
     )
+    log_y = widgets.ToggleButton(
+        value=False,
+        description="Log",
+    )
+
+    def plot_curve(x_axis, curve_a, curve_b, log_y):
+        iterations = result[x_axis]
+        phi_d = result[curve_a]
+        phi_m = result[curve_b]
 
-    def plot_curve(objects):
-
-        inversion = workspace.get_entity(objects)[0]
-        result = None
-        child_names = [k.name for k in inversion.children]
-        if "SimPEG.out" in child_names:
-            result = get_inversion_output(workspace.h5file, objects)
-            iterations = result["iteration"]
-            phi_d = result["phi_d"]
-            phi_m = result["phi_m"]
-
-            ax1 = plt.subplot()
-            ax2 = ax1.twinx()
-            ax1.plot(iterations, phi_d, linewidth=3, c="k")
-            ax1.set_xlabel("Iterations")
-            ax1.set_ylabel(r"$\phi_d$", size=16)
-            ax2.plot(iterations, phi_m, linewidth=3, c="r")
-            ax2.set_ylabel(r"$\phi_m$", size=16)
+        ax1 = plt.subplot()
+        ax2 = ax1.twinx()
+        ax1.plot(iterations, phi_d, linewidth=3, c="k")
```

Comment:
I'm not following you here.  We always plot phi_d and phi_m with the labels set accordingly, but phi_d and phi_m take the data from the dropdown widgets which include iteration, beta as options.  Why would we want to plot iterations/beta vs iterations and label it phi_d/phi_m??  Shouldn't we just drop the widgets and always plot phi_d and phi_m?  Or have a widget that switches between a phid/phim plot and a beta vs iter plot?

Reply:
It was so that users could plot phi_d/phi_m (the classic elbow). But granted, the labels should be changing ...
![image](https://user-images.githubusercontent.com/55204635/216165331-0f47a2d0-e169-4198-9528-4e47f0e81578.png)